Qute namespace extensions without parameters don't work in Quarkus 1.12.0.FINAL

**Describe the bug**
I started giving Quarkus 1.12.0.FINAL a try and potentially found a bug.

When I add a template extensions with a namespace and without any parameters it makes Quarkus 1.12 fail to start. This works fine in previous Quarkus releases and should be supported according to the documentation.

See https://quarkus.io/guides/qute-reference#template_extension_methods

**Expected behavior**
Namespace template extensions work without parameters

**Actual behavior**
Quarkus fails to start and throws the following exception:

```
2021-02-18 09:49:29,197 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.qute.deployment.QuteProcessor#collectTemplateExtensionMethods threw an exception: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base/java.util.Arrays$ArrayList.get(Arrays.java:4351)
	at java.base/java.util.Collections$UnmodifiableList.get(Collections.java:1310)
	at io.quarkus.qute.deployment.QuteProcessor.produceExtensionMethod(QuteProcessor.java:784)
	at io.quarkus.qute.deployment.QuteProcessor.collectTemplateExtensionMethods(QuteProcessor.java:750)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:308)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:182)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:59)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:79)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:377)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:56)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:128)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:81)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:144)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:63)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.qute.deployment.QuteProcessor#collectTemplateExtensionMethods threw an exception: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base/java.util.Arrays$ArrayList.get(Arrays.java:4351)
	at java.base/java.util.Collections$UnmodifiableList.get(Collections.java:1310)
	at io.quarkus.qute.deployment.QuteProcessor.produceExtensionMethod(QuteProcessor.java:784)
	at io.quarkus.qute.deployment.QuteProcessor.collectTemplateExtensionMethods(QuteProcessor.java:750)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.builder.Execution.run(Execution.java:116)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:153)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:306)
	... 9 more
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base/java.util.Arrays$ArrayList.get(Arrays.java:4351)
	at java.base/java.util.Collections$UnmodifiableList.get(Collections.java:1310)
	at io.quarkus.qute.deployment.QuteProcessor.produceExtensionMethod(QuteProcessor.java:784)
	at io.quarkus.qute.deployment.QuteProcessor.collectTemplateExtensionMethods(QuteProcessor.java:750)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

2021-02-18 09:49:29,197 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start hot replacement endpoint to recover from previous Quarkus startup failure
```


**To Reproduce**

Link to a small reproducer (preferably a Maven project if the issue is not Gradle-specific).
https://github.com/wjglerum/quarkus-qute-template-extensions

https://github.com/wjglerum/quarkus-qute-template-extensions/blob/master/src/main/java/nl/wjglerum/qute/QuteTemplateExtension.java#L12

**Configuration**
```properties
# Add your application.properties here, if applicable.

```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
```
Darwin Willems-MBP.i.btp34.nl 20.3.0 Darwin Kernel Version 20.3.0: Thu Jan 21 00:07:06 PST 2021; root:xnu-7195.81.3~1/RELEASE_X86_64 x86_64
```
 - Output of `java -version`: 
 ```
openjdk version "11.0.10" 2021-01-19
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.10+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.10+9, mixed mode)
```
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: `1.12.0.FINAL`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
 ```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /Users/wjglerum/.m2/wrapper/dists/apache-maven-3.6.3-bin/1iopthnavndlasol9gbrbg6bf2/apache-maven-3.6.3
Java version: 11.0.10, vendor: AdoptOpenJDK, runtime: /Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
Default locale: en_NL, platform encoding: UTF-8
OS name: "mac os x", version: "10.16", arch: "x86_64", family: "mac"
```

**Additional context**
(Add any other context about the problem here.)

The generated example code from code.quarkus.io also seems to fail on Quarkus 1.12.0 (figured this out when I removed the Template Extension)

```
2021-02-18 10:29:53,370 WARN  [io.qua.arc.pro.BeanArchives] (build-29) Failed to index org.acme.qute.Quark: Class does not exist in ClassLoader QuarkusClassLoader:Deployment Class Loader
2021-02-18 10:29:53,391 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.qute.deployment.QuteProcessor#processTemplateErrors threw an exception: io.quarkus.qute.TemplateException: Found template problems (4):
[1] Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[2] Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[3] Incorrect expression found: {quark.color.hex}
	- property/method [color] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[4] Incorrect expression found: {quark.flavor.toString.toLowerCase}
	- property/method [flavor] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:42
	at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:178)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.color.hex}
	- property/method [color] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.flavor.toString.toLowerCase}
	- property/method [flavor] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:42
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:308)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:182)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:59)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:79)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:377)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:56)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:127)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:84)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:144)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:63)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.qute.deployment.QuteProcessor#processTemplateErrors threw an exception: io.quarkus.qute.TemplateException: Found template problems (4):
[1] Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[2] Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[3] Incorrect expression found: {quark.color.hex}
	- property/method [color] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[4] Incorrect expression found: {quark.flavor.toString.toLowerCase}
	- property/method [flavor] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:42
	at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:178)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.color.hex}
	- property/method [color] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.flavor.toString.toLowerCase}
	- property/method [flavor] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:42
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more

	at io.quarkus.builder.Execution.run(Execution.java:116)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:153)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:306)
	... 9 more
Caused by: io.quarkus.qute.TemplateException: Found template problems (4):
[1] Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[2] Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[3] Incorrect expression found: {quark.color.hex}
	- property/method [color] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[4] Incorrect expression found: {quark.flavor.toString.toLowerCase}
	- property/method [flavor] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:42
	at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:178)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.color.hex}
	- property/method [color] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.flavor.toString.toLowerCase}
	- property/method [flavor] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:42
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more

2021-02-18 10:29:53,391 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start hot replacement endpoint to recover from previous Quarkus startup failure
2021-02-18 10:30:00,172 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (vert.x-worker-thread-1) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.qute.deployment.QuteProcessor#processTemplateErrors threw an exception: io.quarkus.qute.TemplateException: Found template problems (4):
[1] Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[2] Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[3] Incorrect expression found: {quark.color.hex}
	- property/method [color] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[4] Incorrect expression found: {quark.flavor.toString.toLowerCase}
	- property/method [flavor] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:42
	at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:178)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.color.hex}
	- property/method [color] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.flavor.toString.toLowerCase}
	- property/method [flavor] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:42
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:308)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.reloadExistingApplication(AugmentActionImpl.java:195)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.reloadExistingApplication(AugmentActionImpl.java:59)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.restartApp(IsolatedDevModeMain.java:189)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.restartCallback(IsolatedDevModeMain.java:175)
	at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:255)
	at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$2.handle(VertxHttpHotReplacementSetup.java:62)
	at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$2.handle(VertxHttpHotReplacementSetup.java:52)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:313)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.qute.deployment.QuteProcessor#processTemplateErrors threw an exception: io.quarkus.qute.TemplateException: Found template problems (4):
[1] Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[2] Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[3] Incorrect expression found: {quark.color.hex}
	- property/method [color] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[4] Incorrect expression found: {quark.flavor.toString.toLowerCase}
	- property/method [flavor] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:42
	at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:178)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.color.hex}
	- property/method [color] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.flavor.toString.toLowerCase}
	- property/method [flavor] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:42
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more

	at io.quarkus.builder.Execution.run(Execution.java:116)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:153)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:306)
	... 12 more
Caused by: io.quarkus.qute.TemplateException: Found template problems (4):
[1] Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[2] Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[3] Incorrect expression found: {quark.color.hex}
	- property/method [color] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
[4] Incorrect expression found: {quark.flavor.toString.toLowerCase}
	- property/method [flavor] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:42
	at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:178)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.position}
	- property/method [position] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.color.hex}
	- property/method [color] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:41
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more
	Suppressed: io.quarkus.qute.TemplateException: Incorrect expression found: {quark.flavor.toString.toLowerCase}
	- property/method [flavor] not found on class [org.acme.qute.Quark] nor handled by an extension method
	- at page.qute.html:42
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:158)
		... 10 more

2021-02-18 10:30:00,177 INFO  [io.qua.dep.dev.RuntimeUpdatesProcessor] (vert.x-worker-thread-1) Hot replace total time: 0.205s 
```

--- a/docs/src/main/asciidoc/qute-reference.adoc
+++ b/docs/src/main/asciidoc/qute-reference.adoc
@@ -1249,11 +1249,8 @@ The method name is used to match the property name by default.
 However, it is possible to specify the matching name with `TemplateExtension#matchName()`.
 A special constant - `TemplateExtension#ANY` - may be used to specify that the extension method matches any name.
 It is also possible to match the name against a regular expression specified in `TemplateExtension#matchRegex()`.
-In both cases, a string method parameter is used to pass the property name.
+In both cases, an additional string method parameter must be used to pass the property name.
 If both `matchName()` and `matchRegex()` are set the regular expression is used for matching.
-
-If a namespace is specified the method must declare at least one parameter and the first parameter must be a string. 
-If no namespace is specified the method must declare at least two parameters and the second parameter must be a string. 
 
 .Extension Method Example
 [source,java]

--- a/extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java
+++ b/extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java
@@ -749,7 +749,7 @@ public class QuteProcessor {
                     continue;
                 }
                 if ((namespace == null || namespace.isEmpty()) && method.parameters().isEmpty()) {
-                    // Filter methods with no params for non-namespace extensions
+                    // Filter out methods with no params for non-namespace extensions
                     continue;
                 }
                 if (methods.containsKey(method)) {
@@ -790,7 +790,7 @@ public class QuteProcessor {
             matchRegex = matchRegexValue.asString();
         }
         extensionMethods.produce(new TemplateExtensionMethodBuildItem(method, matchName, matchRegex,
-                method.parameters().get(0), priority, namespace));
+                namespace.isEmpty() ? method.parameters().get(0) : null, priority, namespace));
     }
 
     private void validateInjectExpression(TemplateAnalysis templateAnalysis, Expression expression, IndexView index,

--- a/extensions/qute/deployment/src/test/java/io/quarkus/qute/deployment/extensions/NamespaceTemplateExtensionTest.java
+++ b/extensions/qute/deployment/src/test/java/io/quarkus/qute/deployment/extensions/NamespaceTemplateExtensionTest.java
@@ -39,6 +39,10 @@ public class NamespaceTemplateExtensionTest {
                 engine.parse("{MyEnum:ONE}={MyEnum:one}").render());
         assertEquals("IN_PROGRESS=0",
                 engine.parse("{txPhase:IN_PROGRESS}={txPhase:IN_PROGRESS.ordinal}").render());
+        assertEquals("Quark!",
+                engine.parse("{str:quark}").render());
+        assertEquals("QUARKUS!",
+                engine.parse("{str:quarkus}").render());
     }
 
     @TemplateExtension(namespace = "str")
@@ -52,9 +56,18 @@ public class NamespaceTemplateExtensionTest {
             return new StringBuilder(val).reverse().toString();
         }
 
-        @TemplateExtension(namespace = "str", matchRegex = "foo.*")
+        @TemplateExtension(namespace = "str", matchRegex = "foo.*", priority = 5)
         static String foo(String name, String val) {
             return name + ":" + new StringBuilder(val).reverse().toString();
+        }
+
+        static String quark() {
+            return "Quark!";
+        }
+
+        @TemplateExtension(namespace = "str", matchName = ANY, priority = 1)
+        static String quarkAny(String key) {
+            return key.toUpperCase() + "!";
         }
 
     }

--- a/independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateExtension.java
+++ b/independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateExtension.java
@@ -28,15 +28,10 @@ import java.lang.annotation.Target;
  * <p>
  * By default, the method name is used to match the property name. However, it is possible to specify the matching name with
  * {@link #matchName()}. A special constant - {@link #ANY} - may be used to specify that the extension method matches any name.
- * It is also possible to match the name against a regular expression specified in {@link #matchRegex()}. In both cases, a
- * string method parameter is used to pass the property name.
- * 
+ * It is also possible to match the name against a regular expression specified in {@link #matchRegex()}. In both cases, an
+ * additional string method parameter must be used to pass the property name.
  * <p>
  * If both {@link #matchName()} and {@link #matchRegex()} are set the regular expression is used for matching.
- * 
- * <p>
- * If a namespace is specified the method must declare at least one parameter and the first parameter must be a string. If no
- * namespace is specified the method must declare at least two parameters and the second parameter must be a string.
  * 
  * <pre>
  * {@literal @}TemplateExtension
@@ -88,8 +83,7 @@ public @interface TemplateExtension {
      * <p>
      * Template extension methods that share the same namespace and are declared on the same class are grouped in one resolver
      * and ordered by {@link #priority()}. The first matching extension method is used to resolve an expression. Template
-     * extension
-     * methods declared on different classes cannot share the same namespace.
+     * extension methods declared on different classes cannot share the same namespace.
      * 
      * @return the namespace
      */

--- a/independent-projects/qute/generator/src/main/java/io/quarkus/qute/generator/ExtensionMethodGenerator.java
+++ b/independent-projects/qute/generator/src/main/java/io/quarkus/qute/generator/ExtensionMethodGenerator.java
@@ -81,13 +81,20 @@ public class ExtensionMethodGenerator {
 
     public static void validate(MethodInfo method, List<Type> parameters, String namespace) {
         if (!Modifier.isStatic(method.flags())) {
-            throw new IllegalStateException("Template extension method must be static: " + method);
+            throw new IllegalStateException(
+                    "Template extension method declared on " + method.declaringClass().name() + "  must be static: " + method);
         }
         if (method.returnType().kind() == Kind.VOID) {
-            throw new IllegalStateException("Template extension method must not return void: " + method);
+            throw new IllegalStateException("Template extension method declared on " + method.declaringClass().name()
+                    + " must not return void: " + method);
+        }
+        if (Modifier.isPrivate(method.flags())) {
+            throw new IllegalStateException("Template extension method declared on " + method.declaringClass().name()
+                    + " must not be private: " + method);
         }
         if ((namespace == null || namespace.isEmpty()) && parameters.isEmpty()) {
-            throw new IllegalStateException("Template extension method must declare at least one parameter: " + method);
+            throw new IllegalStateException("Template extension method declared on " + method.declaringClass().name()
+                    + " must declare at least one parameter: " + method);
         }
     }
 
@@ -97,6 +104,7 @@ public class ExtensionMethodGenerator {
         List<Type> parameters = method.parameters();
 
         // Validate the method first
+        // NOTE: this method is never used for namespace extension methods
         validate(method, parameters, null);
         ClassInfo declaringClass = method.declaringClass();
 
@@ -130,8 +138,8 @@ public class ExtensionMethodGenerator {
         }
 
         if (matchRegex != null || matchName.equals(TemplateExtension.ANY)) {
-            // The second parameter must be a string
-            if (parameters.size() < 2 || !parameters.get(1).name().equals(io.quarkus.qute.generator.DotNames.STRING)) {
+            // A string parameter is needed to match the name
+            if (parameters.size() < 2 || !parameters.get(1).name().equals(DotNames.STRING)) {
                 throw new TemplateException(
                         "A template extension method matching multiple names or a regular expression must declare at least two parameters and the second parameter must be string: "
                                 + method);
@@ -461,6 +469,11 @@ public class ExtensionMethodGenerator {
                 }
 
                 boolean matchAnyOrRegex = patternField != null || matchName.equals(TemplateExtension.ANY);
+
+                if (matchAnyOrRegex && paramSize == 0) {
+                    throw new IllegalStateException("Template extension method must declare at least one parameter: " + method);
+                }
+
                 // The real number of evaluated params, i.e. skip the name if matchAny==true
                 int realParamSize = paramSize - (matchAnyOrRegex ? 1 : 0);
 

